codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/EnumGenerator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.go.codegen;
+
+import java.util.Locale;
+import java.util.Map;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.model.shapes.StringShape;
+import software.amazon.smithy.model.traits.EnumConstantBody;
+import software.amazon.smithy.model.traits.EnumTrait;
+import software.amazon.smithy.utils.StringUtils;
+
+/**
+ * Renders enums and their constants.
+ */
+final class EnumGenerator implements Runnable {
+
+    private final SymbolProvider symbolProvider;
+    private final GoWriter writer;
+    private final StringShape shape;
+
+    EnumGenerator(SymbolProvider symbolProvider, GoWriter writer, StringShape shape) {
+        this.symbolProvider = symbolProvider;
+        this.writer = writer;
+        this.shape = shape;
+    }
+
+    @Override
+    public void run() {
+        Symbol symbol = symbolProvider.toSymbol(shape);
+        EnumTrait enumTrait = shape.expectTrait(EnumTrait.class);
+
+        writer.write("type $L string", symbol.getName()).write("");
+
+        writer.write("// Enum values for $L", symbol.getName()).openBlock("const (", ")", () -> {
+            for (Map.Entry<String, EnumConstantBody> entry : enumTrait.getValues().entrySet()) {
+                StringBuilder labelBuilder = new StringBuilder(symbol.getName());
+                String name = entry.getValue().getName().orElse(entry.getKey());
+                for (String part : name.split("\\W")) {
+                    labelBuilder.append(StringUtils.capitalize(part.toLowerCase(Locale.US)));
+                }
+                String label = labelBuilder.toString();
+                writer.write("$L $L = $S", label, symbol.getName(), entry.getKey());
```

Comment:
Would the `Locale.US` prevent non-english Unicode letters from being used in the name of an enum?

Reply:
No, that's just for determining what counts as upper case vs lower case and what the mappings are. E.g. if the locale isn't explicitly set then a capital `I` could become `ı` instead of `i` depending on the locale of the running system.